### PR TITLE
Add support for volatile loads

### PR DESCRIPTION
**Pull request by phate**
This PR does the following:

1. Add LoadOperation and LoadNode class for volatile load operations.
2. Adds conversion support for LLVM -> RVSDG
3. Adds conversion support for RVSDG -> LLVM

### CODE REVIEW

**Review by haved — APPROVED** [review]
Looks good, I think I'm getting a decent understanding of operations vs nodes. I only had one comment where I'm a bit unsure about why the code has curly brackets in a situation where I don't think they do anything, and confuses me a bit.

File: tests/jlm/llvm/backend/llvm/jlm-llvm/LoadTests.cpp [hunk 1]
```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2020 Nico Reißmann <nico.reissmann@gmail.com>
```
[the remark this answers is not in the document]
Comment (phate) [reply tests/jlm/llvm/backend/llvm/jlm-llvm/LoadTests.cpp]: copy & paste error. Fixed now.

File: jlm/llvm/ir/operators/LoadVolatile.hpp [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    LoadVolatileOperation operation(loadedType, 1, alignment);
+    return tac::create(operation, { address, iOState, memoryState });
+  }
+
+private:
+  static std::vector<rvsdg::port>
+  CreateOperandPorts(size_t numStates)
+  {
+    std::vector<rvsdg::port> ports({ PointerType(), iostatetype() });
+    std::vector<rvsdg::port> states(numStates, { MemoryStateType::Create() });
```
[the remark this answers is not in the document]
Comment (phate) [reply jlm/llvm/ir/operators/LoadVolatile.hpp:84]: Yes, they are necessary. The curly braces makes this a port (which takes a type in its constructor) and this port is then added `numStates` times.

[the remark this answers is not in the document]
Comment (phate) [reply jlm/llvm/ir/operators/LoadVolatile.hpp]: Changed from `MemoryStateType::Create()` method to a constructor invocation. That is cheaper.

**Review by haved — APPROVED** [review]

**Pull request merged**